### PR TITLE
Version 1.4.0

### PR DESCRIPTION
**Pull request by antword123**
# Added
- **[FRONTEND][API][Defender]** New Defender Antivirus parser

# Fixed
- **[SECADM]** Removed an outdated secadm rule that could prevent other rules from loading in Apache jail
- **[FRONTEND][API]** Changed the logger of the base API Parser class to `api_parser` instead of `crontab`
- **[FRONTEND][API]** Removed useless imports in almost all api parsers
- **[FRONTEND][API][Imperva]** Removed useless log_downloader module from source
- **[FRONTEND][API]** Fixed many uncatched exceptions in the following parsers : cortex_xdr, forcepoint, harfanglab, imperva, 

# Changed
- **[FRONTEND][API]** Updated all loggers to handle logging in a more generic manner

### CODE REVIEW

File: vulture_os/toolkit/api_parser/proofpoint_tap/proofpoint_tap.py [hunk 1]
```diff
@@ -213,18 +211,15 @@ def execute(self):
                     except ProofpointTAPAPIError as e:
                         error = "API error: {}".format(e)
                 else:
-                    error = "{}".format(e)
+                    error = "{}".format(e) # should be fixed here IMPORTANT
```
Comment (HugoSoszynski) [inline vulture_os/toolkit/api_parser/proofpoint_tap/proofpoint_tap.py:214]:
```suggestion
                    error = "{}".format(e)
```

File: vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def execute(self):
             for filename in self._fetch_files():
                 self.update_lock()
                 file_data = self._download_file(filename)
-                print(file_data)
+                print(file_data) # WTH HERE?
```
Comment (HugoSoszynski) [inline vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py:140]:
```suggestion
                print(file_data)
```

File: vulture_os/toolkit/api_parser/defender/defender.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        current_time = datetime.now(timezone.utc)
+        logs = self.get_logs(logs_from=self.last_api_call, logs_to=current_time)
+
+        # Downloading may take some while, so refresh token in Redis
+        self.update_lock()
+
+        if len(logs) > 0:
+
+            # All alerts are retrieved, reset timezone to now
+            self.frontend.last_api_call = current_time
```
Comment (HugoSoszynski) [inline vulture_os/toolkit/api_parser/defender/defender.py:168]: Make sure no log is duplicated.
Previous call last line == Current call first line.

**Review by HugoSoszynski — CHANGES_REQUESTED** [review]
Could someone please explain the removal of the imperva downloader as I did not find the reason in the commit messages

File: vulture_os/toolkit/api_parser/defender/defender.py (hunk 3 above)
Comment (frikilax) [reply vulture_os/toolkit/api_parser/defender/defender.py]: resolved, thanks for the catch

File: vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py (hunk 2 above)
Comment (frikilax) [reply vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py:140]: This has absolutely no reason to be here
- don't use print() -> use logger instead
- this data could be printed in logs, but in debug only

File: vulture_os/toolkit/api_parser/proofpoint_tap/proofpoint_tap.py (hunk 1 above)
Comment (frikilax) [reply vulture_os/toolkit/api_parser/proofpoint_tap/proofpoint_tap.py:214]: I have no idea why I put that to begin with, I will look into it

Comment (frikilax) [reply vulture_os/toolkit/api_parser/proofpoint_tap/proofpoint_tap.py]: Corrected

File: vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py (hunk 2 above)
Comment (HugoSoszynski) [reply vulture_os/toolkit/api_parser/aws_bucket/aws_bucket.py]: Keeping it as is for now, should be reviewed in the next version

**Review by HugoSoszynski — APPROVED** [review]

**Pull request merged**